lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -389,45 +389,47 @@ sub run_backup {
     my $backupfile;
     my $mimetype;
 
-    if ($request->{backup} eq 'roles'){
-       my @t = localtime(time);
-       $t[4]++;
-       $t[5] += 1900;
-       $t[3] = substr( "0$t[3]", -2 );
-       $t[4] = substr( "0$t[4]", -2 );
-       my $date = "$t[5]-$t[4]-$t[3]";
-
-       $backupfile = $database->backup_globals(
-                      tempdir => $LedgerSMB::Sysconfig::backupdir,
-                         file => "roles_${date}.sql"
-       );
-       $mimetype   = 'text/x-sql';
-    } elsif ($request->{backup} eq 'db'){
-       $backupfile = $database->backup;
-       $mimetype   = 'application/octet-stream';
-    } else {
+    if ($request->{backup} eq 'roles') {
+        my @t = localtime(time);
+        $t[4]++;
+        $t[5] += 1900;
+        $t[3] = substr( "0$t[3]", -2 );
+        $t[4] = substr( "0$t[4]", -2 );
+        my $date = "$t[5]-$t[4]-$t[3]";
+
+        $backupfile = $database->backup_globals(
+            tempdir => $LedgerSMB::Sysconfig::backupdir,
+            file => "roles_${date}.sql"
+        );
+        $mimetype = 'text/x-sql';
+    }
+    elsif ($request->{backup} eq 'db') {
+        $backupfile = $database->backup;
+        $mimetype   = 'application/octet-stream';
+    }
+    else {
         die $request->{_locale}->text('Invalid backup request');
     }
 
     $backupfile or
         die $request->{_locale}->text('Error creating backup file');
 
-    if ($request->{backup_type} eq 'email'){
+    if ($request->{backup_type} eq 'email') {
         # suppress warning of single usage of $LedgerSMB::Sysconfig::...
         no warnings 'once';
```

Comment:
@sbts the `no warnings 'once'` applies to the entire block. The reason we currently need that is because the file refers to that variable in a different namespace exactly once. This is usually the case for Sysconfig:: variables.
The assumption behind the 'once' warnings is that using a variable once is probably a typo.

Now, with the Sysconfig refactoring, where now everything is hidden behind a functional interface, I'm thinking we can probably replace this with a function call instead.

That's outside the scope of this PR, IMO.